src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsnonassociationsapi/resource/PrisonApiResource.kt

```diff
@@ -10,32 +10,55 @@ import org.springframework.web.bind.annotation.GetMapping
 import org.springframework.web.bind.annotation.PathVariable
 import org.springframework.web.bind.annotation.RequestMapping
 import org.springframework.web.bind.annotation.RestController
-import uk.gov.justice.digital.hmpps.hmppsnonassociations.config.ErrorResponse
+import uk.gov.justice.digital.hmpps.hmppsnonassociationsapi.config.ErrorResponse
 import uk.gov.justice.digital.hmpps.hmppsnonassociationsapi.dto.prisonapi.NonAssociationDetails
 import uk.gov.justice.digital.hmpps.hmppsnonassociationsapi.service.NonAssociationsService
 
-// TODO: Check endpoint format/document it
-
 @RestController
 @RequestMapping("/legacy/api", produces = [MediaType.APPLICATION_JSON_VALUE])
 @Tag(name = "Non-Associations", description = "Retrieve non association details from prison-api")
 class PrisonApiResource(
   val nonAssociationsService: NonAssociationsService,
 ) {
-  @GetMapping("/bookings/{bookingId}/non-association-details")
+  @GetMapping("/offenders/{prisonerNumber}/non-association-details")
   @Operation(
-    summary = "Get non-associations by bookingId",
-    description = "TODO",
+    summary = "Get non-associations by Prisoner number",
+    description = "The offender prisoner number",
     responses = [
       ApiResponse(
         responseCode = "200",
-        description = "Returns list of non-associations",
+        description = "Returns non-association details for this prisoner",
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized to access this endpoint",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
       ),
       ApiResponse(
-        responseCode = "400",
-        description = "Incorrect data specified to return list of non-associations",
+        responseCode = "404",
+        description = "Prisoner number not found",
         content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
       ),
+    ],
+  )
+  suspend fun getDetailsByPrisonerNumber(
+    @Schema(description = "The offender prisoner number", example = "A1234BC", required = true)
+    @PathVariable
+    prisonerNumber: String,
+  ): NonAssociationDetails {
+    return nonAssociationsService.getDetails(prisonerNumber)
+  }
+
+  @GetMapping("/bookings/{bookingId}/non-association-details")
```

Comment:
Think we should NOT implement this because WE ARE NOT GOING TO RECORD `bookingId` in our service

Reply:
https://github.com/ministryofjustice/hmpps-non-associations-api/pull/11/commits/a4e2a49ff205cde1ae84ef2d5a45f7851335d5fe